src/cpp/fastdds/xtypes/serializers/idl/dynamic_type_idl.cpp

```diff
@@ -0,0 +1,1086 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/core/Types.hpp>
+#include <fastdds/dds/log/Log.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/detail/dynamic_language_binding.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/DynamicType.hpp>
+
+#include "dynamic_type_idl.hpp"
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+using namespace eprosima::utilities::collections;
+
+constexpr auto TYPE_OPENING = "\n{\n";
+constexpr auto TYPE_CLOSURE = "};\n";
+constexpr auto TAB_SEPARATOR = "    ";
+
+//////////////////////////
+// DYNAMIC TYPE TO TREE //
+//////////////////////////
+
+ReturnCode_t dyn_type_to_tree(
+        const DynamicType::_ref_type& type,
+        const std::string& member_name,
+        TreeNode<TreeNodeType>& node) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    const auto kind = type->get_kind();
+
+    if (kind == TK_STRUCTURE)
+    {
+            // If is struct, the call is recursive.
+            // Create new tree node
+            TreeNode<TreeNodeType> parent(member_name, type->get_name().to_string(), type);
+
+            // Get its base class
+            TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+            ret = type->get_descriptor(type_descriptor);
+
+            if (RETCODE_OK != ret)
+            {
+                return ret;
+            }
```

Comment:
I'm still not satisfied with the amount of traces present for failing scenarios. Not sure to what extent this is relevant. I insist in suggesting to cast to the Impl class, you would be avoiding a big amount of return codes to check.